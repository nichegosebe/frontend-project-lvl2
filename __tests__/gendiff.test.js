import { describe, test, expect } from '@jest/globals';
import { readFileSync } from 'fs';
import { join } from 'path';
import genDiff from '../lib/gendiff';

const getFixturePath = (fileName) => join(process.cwd(), '__tests__', '__fixtures__', fileName);

const fixtureSets = [
  ['file1.json', 'file2.json'],
  ['file1.yaml', 'file2.yaml'],
  ['file1.ini', 'file2.ini'],
  ['file1.ini', 'file2.json'],
  ['file1.yaml', 'file2'],
  ['file1.yaml', 'file2.ini'],
];

const stylishDiffString = readFileSync(getFixturePath('stylish.txt'), 'UTF-8');
const plainDiffString = readFileSync(getFixturePath('plain.txt'), 'UTF-8');
const jsonDiffString = readFileSync(getFixturePath('json.txt'), 'UTF-8');

describe('Tests with different formats', () => {
  test.each(fixtureSets)('%s and %s with default (stylish) formatter', (fileName1, fileName2) => {
    expect(genDiff(getFixturePath(fileName1), getFixturePath(fileName2))).toEqual(
      stylishDiffString,
    );
  });

  test.each(fixtureSets)('%s and %s with plain formatter', (fileName1, fileName2) => {
    expect(genDiff(getFixturePath(fileName1), getFixturePath(fileName2), 'plain')).toEqual(
      plainDiffString,
    );
  });

  test.each(fixtureSets)('%s and %s with json formatter', (fileName1, fileName2) => {
    expect(genDiff(getFixturePath(fileName1), getFixturePath(fileName2), 'json')).toEqual(
      jsonDiffString,
    );
  });
});
