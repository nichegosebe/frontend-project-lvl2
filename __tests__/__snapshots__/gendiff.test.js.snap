// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests with both non-empty fixtures .test with .ini format 1`] = `
"{
    common: {
      + follow: false
        sameArray: 1,2,3
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: {
            key: value
        }
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: too much
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
  - differentArray: c,b,a
  + differentArray: a,b,c
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        fee: 100500
        anotherNullValue: null
        deep: {
            id: {
                number: 45
            }
        }
    }
  - nullValue: null
}"
`;

exports[`Tests with both non-empty fixtures .test with .json format 1`] = `
"{
    common: {
      + follow: false
        sameArray: 1,2,3
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: {
            key: value
        }
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: too much
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
  - differentArray: c,b,a
  + differentArray: a,b,c
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        fee: 100500
        deep: {
            id: {
                number: 45
            }
        }
        anotherNullValue: null
    }
  - nullValue: null
}"
`;

exports[`Tests with both non-empty fixtures .test with .yaml format 1`] = `
"{
    common: {
      + follow: false
        sameArray: 1,2,3
        setting1: Value 1
      - setting2: 200
      - setting3: true
      + setting3: {
            key: value
        }
      + setting4: blah blah
      + setting5: {
            key5: value5
        }
        setting6: {
            doge: {
              - wow: too much
              + wow: so much
            }
            key: value
          + ops: vops
        }
    }
  - differentArray: c,b,a
  + differentArray: a,b,c
    group1: {
      - baz: bas
      + baz: bars
        foo: bar
      - nest: {
            key: value
        }
      + nest: str
    }
  - group2: {
        abc: 12345
        deep: {
            id: 45
        }
    }
  + group3: {
        fee: 100500
        deep: {
            id: {
                number: 45
            }
        }
        anotherNullValue: null
    }
  - nullValue: null
}"
`;
